centreon/src/Core/ResourceAccess/Domain/Model/Rule.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\ResourceAccess\Domain\Model;
+
+use Assert\AssertionFailedException;
+use Centreon\Domain\Common\Assertion\Assertion;
+
+class Rule
+{
+    public const MAX_NAME_LENGTH = 255;
+
+    private ?string $description = null;
+
+    /**
+     * @param int $id
+     * @param string $name
+     * @param bool $isEnabled
+     *
+     * @throws AssertionFailedException
+     */
+    public function __construct(
+        private int $id,
+        private string $name,
+        private bool $isEnabled
+    ) {
+        $shortName = (new \ReflectionClass($this))->getShortName();
+
+        Assertion::positiveInt($id, "{$shortName}::id");
+        Assertion::notEmptyString($this->name, "{$shortName}::name");
+        Assertion::maxLength($this->name, self::MAX_NAME_LENGTH, "{$shortName}::name");
+    }
+
+    /**
+     * @return int
+     */
+    public function getId(): int
+    {
+        return $this->id;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    /**
+     * @param null|string $description
+     *
+     * @return self
+     */
+    public function setDescription(?string $description): self
```

Comment:
Not valid if we consider the object to be immutable, but this subject is under discussion so ok.